app/javascript/dashboard/store/modules/conversations/getters.js

```diff
@@ -50,6 +50,13 @@ const getters = {
   getChatStatusFilter: ({ chatStatusFilter }) => chatStatusFilter,
   getSelectedInbox: ({ currentInbox }) => currentInbox,
   getConvTabStats: ({ convTabStats }) => convTabStats,
+  getNextChatConversation: ({ allConversations, selectedChat }) => {
+    if (allConversations.length <= 1) { return }
```

Comment:
@nithindavid I did some changes and I believe it's more likely what you suggested. I just have one issue

With the current approach, I'm having a difficult that I need help to understand.
- Inside this getter, the `allConversations` state is not being updated and that causes two issues:
  - Resolved conversations are still present on the array, so depending on the order that I solve, it can set a resolved conversation as active.
  - The order is not sorted the same as in the view. I actually think that the order set on `getChatsForTab()` is not persisted in the state, so only available in the view. Maybe we should move the order logic before setting on the state or duplicating it inside the getter?

Thanks again for the help!

Reply:
Let me take a look at this 👍

Reply:
Hey I was able to think of  a working solution with some changes to your PR,
1) We need the old `openNextChat` event which has a `nextChat` as params  in `app/javascript/dashboard/components/ChatList.vue` which should dispatch a `setActiveChat` event.
2) The getter for next chat should look something like
```
getNextChatConversation: _state => {
    const chatList = getters.getAllStatusChats(_state);
    const { selectedChat } = _state;

    const currentIndex = chatList.findIndex(
      conversation => conversation.id === selectedChat.id
    );
    const nextIndex = (currentIndex + 1) % chatList.length;
    return chatList[nextIndex];
  },
```.
3) We need to do a little refactoring by moving the `sorting logic around line 160` in `app/javascript/dashboard/components/ChatList.vue` to `getAllConversations` in `getters.js`
4) Import the `getNextChatConversation` getter in `app/javascript/dashboard/components/buttons/ResolveButton.vue` and call the next chat before  firing the dispatch and emit the `openNextChat` with `nextchat` event when the status resolve is success.

Reply:
@pranavrajs What do you think ?

Reply:
@nithindavid I did some work on what you suggested. Basically just sorting inside the `getters` and getting the status-filtered chats seems to do the work without the need to emit an event.

Could you please take a look when you have some time? https://github.com/chatwoot/chatwoot/pull/757/commits/2ad72c5e1fb2c7992e4fbb261d3cf692e2e062e6